### PR TITLE
Refactor dispatcher cmdline

### PR DESCRIPTION
**Pull request by chicm-ms**
Passthrough experiment settings to dispatcher, remove the unnecessary configuration parsing in nni manager.

Also fix following issue:
recognize customized tuner as builtin tuner if class name is same as one of builtin tuners.

### CODE REVIEW

File: src/sdk/pynni/nni/__main__.py [hunk 1]
```diff
@@ -139,17 +105,14 @@ def main():
             raise
 
 
-def _run_advisor(args):
-    if args.advisor_class_name in AdvisorModuleName:
+def _run_advisor(exp_params):
+    if exp_params.get('advisor').get('builtinAdvisorName') in AdvisorModuleName:
```
Comment (ultmaster) [inline src/sdk/pynni/nni/__main__.py]: If `exp_params.get('advisor')` can be None, I think `get` won't help. Should write `exp_params['advisor']` anyway.

Comment (chicm-ms) [reply src/sdk/pynni/nni/__main__.py]: Here the key 'advisor' must exist, it is OK to use either `exp_params['advisor']` or `exp_params.get('advisor')`, just want to be consistent with other places where key may not exists.

File: src/sdk/pynni/nni/platform/standalone.py [hunk 2]
```diff
@@ -14,7 +14,9 @@
     'send_metric',
 ]
 
-init_standalone_logger()
+if trial_env_vars.NNI_PLATFORM is None:
+    init_standalone_logger()
```
Comment (QuanluZhang) [inline src/sdk/pynni/nni/platform/standalone.py]: this file is for standalone mode, `trial_env_vars.NNI_PLATFORM` must be `None`, right?

Comment (chicm-ms) [reply src/sdk/pynni/nni/platform/standalone.py]: My understanding is yes,  we can see `trial_env_vars.NNI_PLATFORM is None` is used to check standalone mode at other places:
https://github.com/microsoft/nni/blob/master/src/sdk/pynni/nni/trial.py#L111
https://github.com/microsoft/nni/blob/master/src/sdk/pynni/nni/platform/__init__.py#L6

Comment (chicm-ms) [reply src/sdk/pynni/nni/platform/standalone.py]: Good catch, the problem is on environment variables, NNI_PLATFORM is an env var for trial not for dispatcher,  I will fix it in another PR.

**Pull request merged**